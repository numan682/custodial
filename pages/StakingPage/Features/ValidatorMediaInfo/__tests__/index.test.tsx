import { render, screen } from '@testing-library/react'
import * as React from 'react'

import { ValidatorMediaInfo } from '..'

describe('<ValidatorMediaInfo />', () => {
  it('should match snapshot', () => {
    const component = render(
      <ValidatorMediaInfo
        mediaInfo={{
          logotype: 'https://s3.amazonaws.com/s.jpg',
          website_link: 'https://s.f',
          twitter_acc: 'https://twitter.com/s',
          email_address: 's@s.f',
        }}
      />,
    )
    expect(component.baseElement).toMatchSnapshot()
  })

  it('should display a link if the address is valid', () => {
    render(<ValidatorMediaInfo mediaInfo={{ website_link: 'http://www.google.fr/hello' }} />)
    const el = screen.getByRole('link')
    expect(el).toHaveAttribute('href', 'http://www.google.fr/hello')
  })

  it('should not display a link if the address is invalid', () => {
    // eslint-disable-next-line no-script-url
    render(<ValidatorMediaInfo mediaInfo={{ website_link: 'javascript:alert(1)' }} />)
    const el = screen.queryByRole('link')
    expect(el).toBeNull()
  })

  it('should display an email if the address is valid', () => {
    render(<ValidatorMediaInfo mediaInfo={{ email_address: 'test@test.com' }} />)
    const el = screen.getByRole('link')
    expect(el).toHaveAttribute('href', 'mailto:test@test.com')
  })

  it('should not display the email if the address is dangerous', () => {
    render(<ValidatorMediaInfo mediaInfo={{ email_address: 'test@test.com?attach=dangerouspayload' }} />)
    const el = screen.queryByRole('link')
    expect(el).toBeNull()
  })
})
