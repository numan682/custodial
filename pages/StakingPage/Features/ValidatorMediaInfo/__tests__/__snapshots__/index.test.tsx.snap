// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidatorMediaInfo /> should match snapshot 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c3 *[stroke*='#'],.c3 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*='#'],.c3 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c2:hover {
  background-color: rgba(221,221,221,0.4);
  color: #000000;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c2:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible) > circle,.c2:focus:not(:focus-visible) > ellipse,
.c2:focus:not(:focus-visible) > line,.c2:focus:not(:focus-visible) > path,
.c2:focus:not(:focus-visible) > polygon,.c2:focus:not(:focus-visible) > polyline,
.c2:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    width: auto;
    height: 12px;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <img
        class="logotype-big"
        src="https://s3.amazonaws.com/s.jpg"
      />
      <div
        class="c1"
      />
      <div
        class="c0"
      >
        <a
          class="c2"
          href="mailto:s@s.f"
          rel="noopener"
          target="_blank"
        >
          <svg
            aria-label="MailOption"
            class="c3"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 4h22v16H1V4zm0 1 11 8.5L23 5"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </a>
        <a
          class="c2"
          href="https://s.f"
          rel="noopener"
          target="_blank"
        >
          <svg
            aria-label="Home"
            class="c3"
            viewBox="0 0 24 24"
          >
            <path
              d="m1 11 11-9 11 9m-8 12v-8H9v8m-5 0V9m16 14V9"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </a>
        <a
          class="c2"
          href="https://twitter.com/s"
          rel="noopener"
          target="_blank"
        >
          <svg
            aria-label="Twitter"
            class="c3"
            viewBox="0 0 24 24"
          >
            <path
              d="M24 4.309a9.83 9.83 0 0 1-2.828.775 4.94 4.94 0 0 0 2.165-2.724 9.865 9.865 0 0 1-3.127 1.196 4.925 4.925 0 0 0-8.39 4.49A13.974 13.974 0 0 1 1.671 2.9a4.902 4.902 0 0 0-.667 2.476c0 1.708.869 3.216 2.191 4.099A4.936 4.936 0 0 1 .964 8.86v.06a4.926 4.926 0 0 0 3.95 4.829 4.964 4.964 0 0 1-2.224.085 4.93 4.93 0 0 0 4.6 3.42 9.886 9.886 0 0 1-6.115 2.107c-.398 0-.79-.023-1.175-.068a13.945 13.945 0 0 0 7.548 2.212c9.057 0 14.009-7.503 14.009-14.01 0-.213-.005-.425-.014-.636A10.012 10.012 0 0 0 24 4.309"
              fill="#1DA1F2"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</body>
`;
